### DISCUSSION TITLE
Jtx connected to DavX, but creates only Local elements

### DISCUSSION THREAD

**Discussion by unalignedcoder** [TechbeeAT/jtxBoard#252] [General]
I am just trying this, so bear with me.

I have a working DavX, featuring multiple calendars, tasks, contacts etc.
JtxBoard seems to have acknowledged the presence of this DavX connection. 

Yet **any task, journal or note entry I want to create from within Jtx can only be "Local"**. Clicking on the dropdown menu where it says "Local" doesn't do anything.

The app seems to retrieve existing tasks via DavX correctly, but that's the extent of it. Anything else is done locally.

If I go into "Collections" and select "Add remote collection", DavX is opened, where my connected account is already present.
I understand "Add remote collection" is really just an invitation to create a new account with DavX. Still, in this case, it doesn't work properly, because, though the account is already there, everything can only be saved locally.

How to fix this?

**Comment by patrickunterwegs** [comment]
Please check the following link if it contains your problem/solution:
https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong

Which version of jtx Board are you using?

**Reply by unalignedcoder** [reply]
I am using v 2.02.00-rc12. 
Davx is connected to Nextcloud. 
It seems I do am able to create tasks remotely, but journals and notes only locally. 
Is this because Nextcloud calendar doesn't support VTODO or VJOURNAL? Seems odd.
I could not find anywhere the "symbols next to the collection" referred to in that troubleshoot.

**Reply by patrickunterwegs** [reply]
Every collection itself has defined which components it supports. If you have no collection that supports journals/notes, you have to create a new one (e.g. from DAVx5) and make sure the journals/notes checkbox is activated!

**Reply by unalignedcoder** [reply]
I don't really want to create a new collection. 
I'd rather use the Nextcloud calendar, tasks and contacts I already have already setup in DavX. 
Can you please elaborate as to why this doesn't simply work out of the box?

**Reply by patrickunterwegs** [reply]
I don't know why Nextcloud doesn't create collections with journals/notes support by default. This is a question that you should address with Nextcloud.

**Reply by unalignedcoder** [reply]
>address with nextcloud

I would never do that as I don't enjoy being ignored...
Just out of curiosity, what kind of webdav servers do you know that work out of the box remotely with jtx?

**Reply by patrickunterwegs** [reply]
Well, the thing is that jtx Board is the only app (together with 2 Linux desktop applications) that supports the VJOURNAL component which would take journals and notes.
When a new collection is created a you don't explicitly specify which components it should support (VEVENT, VTODO, VJOURNAL), per definition it should support all of them. BUT in fact most servers would only support VTODO and VEVENT in this case (which is actually a bug on most servers). Unless you create a collection with all components explicitly. 

So I'm afraid there is no way around creating a new collection. But maybe you can create a new one with all components and just move all your entries there.

**Reply by unalignedcoder** [reply]
But that would be only local, no?

**Reply by unalignedcoder** [reply]
When I click on "create a new collection" it sends me to davx. The only davx/webdav account I have is with nextcloud. Are you suggesting I should redo my davx connection to nextcloud? Wouldn't that end up having the same limitations?

**Reply by patrickunterwegs** [reply]
Within DAVx5 you can create a new remote collection on your server, the menu actually says "Create new calendar":

![Screenshot_20230203_093903_DAVx.jpg](https://user-images.githubusercontent.com/72232737/216552612-b6c89abb-9d08-460e-9d96-5fc8e93ed730.jpg)

![Screenshot_20230203_093921_DAVx.jpg](https://user-images.githubusercontent.com/72232737/216552635-4b16fb61-8bcb-4d7f-a62d-724fdf24835d.jpg)

**Reply by unalignedcoder** [reply]
Oh. I finally see what you mean... So this is happening within the davx interface, and those icons referring to the properties of the collection are to be found within davx. okay. Was not clear from your troubleshooting guide.
Well, I see that a new calendar would appear to include journal functionality (this is good), while the old ones do not (this is bad), even though they are all on the same server (this is odd).

I suppose there is no way to edit the properties of the existing calendars to include journaling abilities?

**Comment by unalignedcoder** [comment]
Uh. Interesting. This is one column in the `oc_calendars` table of Nextcloud database:

![2023-02-03 14_51_55-Technocracy __ tuscoss com _ localhost _ jagonoja_nextcloud _ oc_calendars _ php](https://user-images.githubusercontent.com/16850566/216555036-b071e779-17be-4eae-b3e8-e80fa75b86d4.png)

It seems editing those entries might enable Journaling for all calendars...

[Edit:] I did enable journaling for an old calendar this way (editing the Nextcloud table above via phpmyadmin). **However, when creating or editing journal entries in Jtx, I still cannot select anything else but "Local".**

[Edit 2]: Resyncing Jtx with Davx solved this problem. **I can now save Journals and Notes remotely.**

Feel free to mark this as a solution.

**Reply by patrickunterwegs** [reply]
Thanks for the update, I'm glad it's sorted out now 😉
As far as I know adding the VJOURNAL value in the database of Nextcloud is anyway the only way to update the collection! So that should be safe 👍

**Comment by nilskamm** [comment]
Hey! 
I've got a similar problem. I can only create local tasks.
Tasks from my Nextcloud, synched with DAVx5 are shown in jtx and when i mark a task as completed, the new status also gets synched to my nextcloud. But when creating a new task i can not choose something else than local.
I only use tasks. Not interested in notes / journals at the moment. 

jtx Board Version: 2.02.02-rc5.gplay
In DAVx5 i see the entries exactly as they are in nextcloud: 3 tasklists and 1 calendar+tasklist

Any idea how i can fix that? I already checked the "Troubles with the synchronization?" Link.
Thanks a lot!

**Reply by patrickunterwegs** [reply]
Hey @nilskamm , 
In the Google Play Version the creation of remote entries is only possible after you buy the Pro upgrade. Have you done this already or was there maybe any issue with the upgrade?

**Reply by nilskamm** [reply]
Thank you! 
Just installed it yesterday for the first time to check it out. 
Didn't know, this is a Pro feature and unlike the other options the app didn't tell me. 
I will check it out further and then take a look on the Pro upgrade.

**Reply by patrickunterwegs** [reply]
Alright, sure! I'll also check how to improve the screens to make clear why you can't select a remote collection!
If you want to be sure that everything works as expected, you can also take the open source version from f-Droid!

**Reply by nilskamm** [reply]
Thank you for that information. I will check it out!

**Comment by oakbrad** [comment]
I seem to be having a similar issue. I have a baikal server and created remote calendars on it with DavX5 as detailed above, with all the CalDAV options enabled. I have that account set up in DavX and it syncs to my other apps just fine. I've gone through the "Troubles with synchronization" link and enabled all the permissions and set JTX as the Integration.

In JTX I do not see any of my tasks from my DavX account and can only add local items. "Please add a local collection or synchronise a remote collection that supports this module." I click the "Sync with DavX5" menu and it says "Congratulations, we have detected DavX" Clicking sync on this page does nothing, and the "Add account in DavX" just brings me to the account I've already added in DavX that I want to sync. Same with "Synchronise Remote collection" in the Collections menu.

Am I missing something obvious?

**Reply by patrickunterwegs** [reply]
Hey @oakbrad !
Which versions of jtx Board and DAVx5 are you using?
Is there a red sync symbol at the bottom of jtx Board in the list view? 
Are you sure you have set jtx Board as the preferred "Tasks app" in the settings in DAVx5?
Could you maybe also provide the debug info from DAVx5 (from the settings menu)?

**Comment by mateimm** [comment]
Hello. I have installed and am able to connect to caldav. How do i test if the sync succeeded thank you.

**Reply by patrickunterwegs** [reply]
Hello @mateimm , which versions of jtx Board and DAVx5 are you using?
Basically you can see the collections that you have selected in DAVx5 in the collections menu or entries will appear, if there are any tasks, notes or journals in the collection.

**Reply by mateimm** [reply]
@patrickunterwegs I am using the attached versions. I am not sure where I could find the collections menu in DAVx5 you mentioned. 
![Screenshot_20230512-222635](https://github.com/TechbeeAT/jtxBoard/assets/26927895/a4d1bcea-7d9f-4796-a573-a665017c7098)
![Screenshot_20230512-222654](https://github.com/TechbeeAT/jtxBoard/assets/26927895/d7c50b61-60c8-46ce-adc8-175e27b7ff9e)

**Reply by patrickunterwegs** [reply]
Thanks. 
Well, you will find the collections menu point when you open the menu on the top left and click on collections. 

Please have a look here: 
https://jtx.techbee.at/sync-with-davx5
and here for more information and possible troubleshooting about the sync: 
https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong

**Reply by mateimm** [reply]
Thank you @patrickunterwegs for those links and quick answers. I have already checked them before coming to write here. 
I have no collections in the DAVx5 menu. Please see below.
![Screenshot_20230512-223608](https://github.com/TechbeeAT/jtxBoard/assets/26927895/0f1e9bda-3f8e-4e21-9631-4c837d437e20)

**Reply by patrickunterwegs** [reply]
You are checking in DAVx5, but you want to find the entries in jtx Board, so you should check there!

**Reply by mateimm** [reply]
I see now. I see the number of items synchronized. Thank you.
![Screenshot_20230512-225045_2_1](https://github.com/TechbeeAT/jtxBoard/assets/26927895/ad51843b-b28b-4fc4-b906-03356088ca43)
